crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -366,6 +369,20 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                     continue 'FORK;
                 };
                 state.next = highest_priority_pkg;
+                let url = state.next.name().and_then(|name| state.fork_urls.get(name));
+
+                // Consider:
+                // ```toml
+                // dependencies = [
+                //   "iniconfig == 1.1.1 ; python_version < '3.12'",
+                //   "iniconfig @ https://files.pythonhosted.org/packages/ef/a6/62565a6e1cf69e10f5727360368e451d4b7f58beeac6173dc9db836a5b46/iniconfig-2.0.0-py3-none-any.whl ; python_version >= '3.12'",
+                // ]
+                // ```
+                // In the `python_version < '3.12'` case, we haven't pre-visited `iniconfig` yet,
+                // since we weren't sure whether it might also be a URL requirement when
+                // transforming the requirements. For that case, we do another request here
+                // (idempotent due to caching).
+                self.request_package(&state.next, url, &request_sink)?;
```

Comment:
Is this only necessary if `url` is `None`?

Reply:
We need it if `fork_urls` return `None` and `Urls` return something, that is the case when it's ambiguous. I didn't add that check specifically because `request_package` already checks if the request was previously registered